dataStructures/queues/queue.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"fmt"
+)
+
+// Queue data structure, it will be a list of Orders
+type Queue []*Order
+
+// Order data type
+type Order struct {
+	priority     int
+	quantity     int
+	product      string
+	customerName string
+}
+
+// New method creates a new order with required properties
+func (order *Order) New(priority, quantity int, product, customerName string) {
+	order.priority = priority
+	order.quantity = quantity
+	order.product = product
+	order.customerName = customerName
+}
+
+// Add method takes an order and adds it to the queue based on priority
+func (queue *Queue) Add(order *Order) {
```

Comment:
`func (q *Queue)`

The variable used when defining methods is called the "receiver". It should be very short and the same across all methods. The reason that it can be short is because when someone is reading the code they know from the context that it is a method on Queue. [More info](https://github.com/golang/go/wiki/CodeReviewComments#receiver-names).

Reply:
@aschlosberg Yeah, well understood.